tests/test_outputs/test_form_output_prep.py

```diff
@@ -0,0 +1,328 @@
+            [39900000404, 2, 202212, 'AA', 0.0, 0.0, 100.0, 'Yes', 6.0, 6, 40],
+            [39900000408, 0, 202212, 'AA', np.nan, np.nan, 100.0, 'No', 1.0, 6, 40],
+            [39900000408, 1, 202212, 'AA', 0.0, 1222.0, 100.0, 'No', 1.0, 6, 40], # 604 No and 211 > 0
+            [39900000408, 2, 202212, 'AA', 0.0, 0.0, 100.0, 'No', 1.0, 6, 40],
+            [39900000576, 1, 202212, 'AA', 0.0, 0.0, 100.0, np.nan, 1.0, 1, 40],
+            [39900000960, 0, 202212, 'AA', np.nan, np.nan, np.nan, 'No', 1.0, 1, 40],
+            [39900000960, 1, 202212, 'AA', np.nan, 0.0, np.nan, 'No', 1.0, 1, 40],
+            [39900001029, 1, 202212, 'I', 119376.5, 244543.0, 100.0, np.nan, 1.0, 1, 14]
+        ]
+        estimated_df = pd.DataFrame(data=data, columns=columns)
+        return estimated_df
+    
+
+    @pytest.fixture(scope="function")
+    def weighted_df(self) -> pd.DataFrame:
+        """'weighted_df' input for form_output_prep.
+        
+        NOTE: This is a subset of columns for testing purposes
+        """
+        columns = [
+            "reference", 
+            "instance", 
+            "period",
+            "201", #pg
+            "202",
+            "211",
+            "602", 
+            "604", 
+            "a_weight", 
+            "formtype",
+            "pg_numeric"
+        ]
+        data = [
+            [39900000404, 1, 202212, 'AA', 467705.0, 688000.0, 100.0, 'Yes', 6.0, 6, 40],
+            [39900000404, 2, 202212, 'AA', 0.0, 0.0, 100.0, 'Yes', 6.0, 6, 40],
+            [39900000408, 1, 202212, 'AA', 0.0, 0.0, 100.0, 'No', 1.0, 6, 40],
+            [39900000408, 2, 202212, 'AA', 0.0, 0.0, 100.0, 'No', 1.0, 6, 40],
+            [39900000576, 1, 202212, 'AA', 0.0, 0.0, 100.0, np.nan, 1.0, 1, 40],
+            [39900000960, 1, 202212, 'AA', np.nan, 0.0, np.nan, 'No', 1.0, 1, 40],
+            [39900001029, 1, 202212, 'I', 119376.5, 244543.0, 100.0, np.nan, 1.0, 1, 14],
+            [39900001031, 1, 202212, 'U', 0.0, 0.0, 100.0, 'No', 49.0, 6, 30],
+            [39900001031, 2, 202212, 'U', 0.0, 0.0, 100.0, 'No', 49.0, 6, 30],
+            [39900001032, 1, 202212, 'I', 119376.5, 244543.0, 100.0, np.nan, 1.0, 1, 14]
+        ]
+        weighted_df = pd.DataFrame(data=data, columns=columns)
+        return weighted_df
+    
+
+    @pytest.fixture(scope="function")
+    def ni_full_responses(self) -> pd.DataFrame:
+        """'ni_full_responses' input for form_output_prep.
+        
+        NOTE: This is a subset of columns for testing purposes
+        """
+        columns = [
+            "reference", 
+            "instance",
+            "period_year",
+            "foc",
+            "407",
+            "201", #pg
+            "rusic" #sic
+        ]
+        data = [
+            [39900001577, 1, 2021, 'GB ', 0.72, np.nan, 10130],
+            [39900006060, 1, 2021, 'GB ', 0.6, np.nan, 20120],
+            [39900008752, 1, 2021, 'GB ', 0.72, np.nan, 46420],
```

Comment:
NI data has occasional entries for col 201, in integer form. (you can see this in the staged_NI_full_responses csv in the northern_ireland folder of the intermediate outputs). Please could you include this case in your test?